{"rendered": {"message": {"raw": "An attempt to fix issue #45.\n\nAlthough I don't see why the Qt model or dataframe might sometimes not update,\nI noticed that the calls to update_row with updated_row_data provided were\n*not* actually updating the Qt model, because the check for whether a column\nwas in the updated_row_data dictionary was comparing a column tuple padded with\nempty strings to one not padded with empty strings. Further, the call to update\nthe progress percentage was updating *all* qt model items for the row.\n\nNow, the two things are fixed - the updated_row_data does actually update\nthe qt model items, and the update of the status percentage does not.\n\nThere is no reason to believe this will fix issue #45 other than it touching\ncode that could be relevant and was not working as intended. I still don't\nsee specifically what could cause issue #45.\n\nAlso, the updated data dict sent back to lyse from analysis routines should\nonly include items under the 'results/' group in the HDF5 file. So this fixes\nthat too.", "markup": "markdown", "html": "<p>An attempt to fix issue <a href=\"#!/labscript_suite/lyse/issues/45/dataframe-stopped-updating-after-multishot\" rel=\"nofollow\" title=\"Dataframe stopped updating after multishot routine error\" class=\"ap-connect-link\"><s>#45</s></a>.</p>\n<p>Although I don't see why the Qt model or dataframe might sometimes not update,<br />\nI noticed that the calls to update_row with updated_row_data provided were<br />\n<em>not</em> actually updating the Qt model, because the check for whether a column<br />\nwas in the updated_row_data dictionary was comparing a column tuple padded with<br />\nempty strings to one not padded with empty strings. Further, the call to update<br />\nthe progress percentage was updating <em>all</em> qt model items for the row.</p>\n<p>Now, the two things are fixed - the updated_row_data does actually update<br />\nthe qt model items, and the update of the status percentage does not.</p>\n<p>There is no reason to believe this will fix issue <a href=\"#!/labscript_suite/lyse/issues/45/dataframe-stopped-updating-after-multishot\" rel=\"nofollow\" title=\"Dataframe stopped updating after multishot routine error\" class=\"ap-connect-link\"><s>#45</s></a> other than it touching<br />\ncode that could be relevant and was not working as intended. I still don't<br />\nsee specifically what could cause issue <a href=\"#!/labscript_suite/lyse/issues/45/dataframe-stopped-updating-after-multishot\" rel=\"nofollow\" title=\"Dataframe stopped updating after multishot routine error\" class=\"ap-connect-link\"><s>#45</s></a>.</p>\n<p>Also, the updated data dict sent back to lyse from analysis routines should<br />\nonly include items under the 'results/' group in the HDF5 file. So this fixes<br />\nthat too.</p>", "type": "rendered"}}, "hash": "106e07ca8072c32c55002c34913c0f2c518f38ba", "repository": {"links": {"self": {"href": "data/repositories/labscript_suite/lyse.json"}, "html": {"href": "#!/labscript_suite/lyse"}, "avatar": {"href": "data/bytebucket.org/ravatar/{55eebdfe-43d1-4ae8-9049-50c55b295397}ts=249921"}}, "type": "repository", "name": "lyse", "full_name": "labscript_suite/lyse", "uuid": "{55eebdfe-43d1-4ae8-9049-50c55b295397}"}, "links": {"self": {"href": "data/repositories/labscript_suite/lyse/commit/106e07ca8072c32c55002c34913c0f2c518f38ba.json"}, "comments": {"href": "data/repositories/labscript_suite/lyse/commit/106e07ca8072c32c55002c34913c0f2c518f38ba/comments_page=1.json"}, "patch": {"href": "https://api.bitbucket.org/2.0/repositories/labscript_suite/lyse/patch/106e07ca8072c32c55002c34913c0f2c518f38ba"}, "html": {"href": "#!/labscript_suite/lyse/commits/106e07ca8072c32c55002c34913c0f2c518f38ba"}, "diff": {"href": "https://api.bitbucket.org/2.0/repositories/labscript_suite/lyse/diff/106e07ca8072c32c55002c34913c0f2c518f38ba"}, "approve": {"href": "https://api.bitbucket.org/2.0/repositories/labscript_suite/lyse/commit/106e07ca8072c32c55002c34913c0f2c518f38ba/approve"}, "statuses": {"href": "data/repositories/labscript_suite/lyse/commit/106e07ca8072c32c55002c34913c0f2c518f38ba/statuses_page=1.json"}}, "author": {"raw": "chrisjbillington", "type": "author", "user": {"display_name": "Chris Billington", "uuid": "{e363c5a9-5075-4656-afb5-88bd6a6dceeb}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D"}, "html": {"href": "https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/9238baf7300c41c0e7294db922899e6ad=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsCB-1.png"}}, "nickname": "cbillington", "type": "user", "account_id": "557058:cbf1bc43-1dc2-477b-9e25-1a8f40fd7ee3"}}, "summary": {"raw": "An attempt to fix issue #45.\n\nAlthough I don't see why the Qt model or dataframe might sometimes not update,\nI noticed that the calls to update_row with updated_row_data provided were\n*not* actually updating the Qt model, because the check for whether a column\nwas in the updated_row_data dictionary was comparing a column tuple padded with\nempty strings to one not padded with empty strings. Further, the call to update\nthe progress percentage was updating *all* qt model items for the row.\n\nNow, the two things are fixed - the updated_row_data does actually update\nthe qt model items, and the update of the status percentage does not.\n\nThere is no reason to believe this will fix issue #45 other than it touching\ncode that could be relevant and was not working as intended. I still don't\nsee specifically what could cause issue #45.\n\nAlso, the updated data dict sent back to lyse from analysis routines should\nonly include items under the 'results/' group in the HDF5 file. So this fixes\nthat too.", "markup": "markdown", "html": "<p>An attempt to fix issue <a href=\"#!/labscript_suite/lyse/issues/45/dataframe-stopped-updating-after-multishot\" rel=\"nofollow\" title=\"Dataframe stopped updating after multishot routine error\" class=\"ap-connect-link\"><s>#45</s></a>.</p>\n<p>Although I don't see why the Qt model or dataframe might sometimes not update,<br />\nI noticed that the calls to update_row with updated_row_data provided were<br />\n<em>not</em> actually updating the Qt model, because the check for whether a column<br />\nwas in the updated_row_data dictionary was comparing a column tuple padded with<br />\nempty strings to one not padded with empty strings. Further, the call to update<br />\nthe progress percentage was updating <em>all</em> qt model items for the row.</p>\n<p>Now, the two things are fixed - the updated_row_data does actually update<br />\nthe qt model items, and the update of the status percentage does not.</p>\n<p>There is no reason to believe this will fix issue <a href=\"#!/labscript_suite/lyse/issues/45/dataframe-stopped-updating-after-multishot\" rel=\"nofollow\" title=\"Dataframe stopped updating after multishot routine error\" class=\"ap-connect-link\"><s>#45</s></a> other than it touching<br />\ncode that could be relevant and was not working as intended. I still don't<br />\nsee specifically what could cause issue <a href=\"#!/labscript_suite/lyse/issues/45/dataframe-stopped-updating-after-multishot\" rel=\"nofollow\" title=\"Dataframe stopped updating after multishot routine error\" class=\"ap-connect-link\"><s>#45</s></a>.</p>\n<p>Also, the updated data dict sent back to lyse from analysis routines should<br />\nonly include items under the 'results/' group in the HDF5 file. So this fixes<br />\nthat too.</p>", "type": "rendered"}, "participants": [], "parents": [{"hash": "743d4ad4fc8a36153306e52da76346f9e0282bbd", "type": "commit", "links": {"self": {"href": "data/repositories/labscript_suite/lyse/commit/743d4ad4fc8a36153306e52da76346f9e0282bbd.json"}, "html": {"href": "#!/labscript_suite/lyse/commits/743d4ad4fc8a36153306e52da76346f9e0282bbd"}}}], "date": "2019-08-06T20:06:41+00:00", "message": "An attempt to fix issue #45.\n\nAlthough I don't see why the Qt model or dataframe might sometimes not update,\nI noticed that the calls to update_row with updated_row_data provided were\n*not* actually updating the Qt model, because the check for whether a column\nwas in the updated_row_data dictionary was comparing a column tuple padded with\nempty strings to one not padded with empty strings. Further, the call to update\nthe progress percentage was updating *all* qt model items for the row.\n\nNow, the two things are fixed - the updated_row_data does actually update\nthe qt model items, and the update of the status percentage does not.\n\nThere is no reason to believe this will fix issue #45 other than it touching\ncode that could be relevant and was not working as intended. I still don't\nsee specifically what could cause issue #45.\n\nAlso, the updated data dict sent back to lyse from analysis routines should\nonly include items under the 'results/' group in the HDF5 file. So this fixes\nthat too.", "type": "commit", "git_hash": "243198937969ba47bf4f55adacc1be66a3e307fa", "tags": null, "branches": "attempted-bugfix"}